basis/http/client/client.factor

```diff
@@ -111,15 +142,73 @@ SYMBOL: redirects
     "transfer-encoding" header "chunked" =
     [ read-chunked ] [ each-block ] if ; inline
 
+: request-socket-endpoints ( request -- physical logical )
+    [ proxy-url>> ] [ url>> ] bi [ or ] keep ;
+
 : <request-socket> ( -- stream )
-    request get url>> url-addr ascii <client> drop
+    request get request-socket-endpoints [ url-addr ] bi@
+    remote-address set ascii <client> local-address set
     1 minutes over set-timeout ;
 
+: https-tunnel? ( request -- ? )
+    [ proxy-url>> ] [ url>> protocol>> "https" = ] bi and ;
```

Comment:
Yeah, I'm not sure what's best. What I did was:
- f means without proxy; a url means to try with proxy. So by default it has an empty url, so by default it tries to use a proxy.
- the proxy url is then used with derive-url from the proxy variables. So if you setup the variable and leave the default empty url, you'll get a copy from the variables. But if you fill in some slots of the default url, you'll override these values from the variables.

And at the end, if the resulting proxy-url doesn't have a host, there is no proxy. So without variables and with the default empty url, it doesn't use a proxy.